modules/cudf/src/table/arrow.cpp

```diff
@@ -57,4 +58,41 @@ Napi::Value Table::to_arrow(Napi::CallbackInfo const& info) {
 
   return Napi::Uint8Array::New(env, buffer.ByteLength(), buffer, 0);
 }
+
+Napi::Value Table::from_arrow(Napi::CallbackInfo const& info) {
+  CallbackArgs args{info};
+  auto env = info.Env();
+
+  auto device_manager = arrow::cuda::CudaDeviceManager::Instance().ValueOrDie();
+  auto context        = device_manager->GetContext(Device::active_device_id()).ValueOrDie();
+
+  Span<uint8_t> span = args[0];
+  auto buffer        = std::make_shared<arrow::cuda::CudaBuffer>(
+    span.data(), span.size(), context, false, IpcMemory::IsInstance(args[0]));
+
+  auto buffer_reader  = new arrow::cuda::CudaBufferReader(buffer);
+  auto message_reader = CudaMessageReader::Open(buffer_reader, nullptr);
+  auto stream_reader =
+    arrow::ipc::RecordBatchStreamReader::Open(std::move(message_reader)).ValueOrDie();
+
+  std::shared_ptr<arrow::Table> arrow_table{};
+  auto status = stream_reader->ReadAll(&arrow_table);
+
+  if (!status.ok()) { NAPI_THROW(Napi::Error::New(env, status.message())); }
+
+  auto output = Napi::Object::New(env);
+
+  auto fields = stream_reader->schema()->fields();
+  auto names  = Napi::Array::New(env, fields.size());
+  for (std::size_t i = 0; i < fields.size(); ++i) { names.Set(i, fields[i]->name()); }
```

Comment:
Is there a `std` operations for this? `transform`? Regardless, this still looks clean.

Reply:
We would need an NAPI approach that didn't require the iteration index, or `std` operation that provides one (i.e. like Python's `enumerate`) but AFAIK I don't think either exists.

Reply:
It'd be nice if `Napi::Array` had an iterator similar to https://github.com/nodejs/node-addon-api/pull/930